### PR TITLE
feat(collection): create collection page [MS-954]

### PR DESCRIPTION
**Pull request by gajda-w**
I want to merge this change because...

# Pull Request Checklist

- [x] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/(main)/collections/[slug]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      },
+    },
+  });
+
+  const collection = getCollectionResult.data?.results;
+
+  return {
+    title: collection?.seoTitle || collection?.name,
+    description: collection?.description,
+  };
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/app/[locale]/(main)/collections/[slug]/page.tsx:56]: You can also extend this by adding a `openGraph` image property with the background image

File: packages/infrastructure/src/public/saleor/collection/types.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { type Collection } from "@nimara/domain/objects/Collection";
+import { type AsyncResult } from "@nimara/domain/objects/Result";
+
+import { type FetchOptions } from "#root/graphql/client";
+
+export type SaleorCollectionServiceConfig = {
+  apiURI: string;
+};
```
Comment (GrzegorzDerdak) [inline packages/infrastructure/src/public/saleor/collection/types.ts:8]:
```suggestion
export type SaleorCollectionServiceConfig = {
  apiURI: string;
  logger: Logger;
};
```

File: packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+import { CollectionDetailsQueryDocument } from "../graphql/queries/generated";
+import {
+  type GetCollectionDetailsInfra,
+  type SaleorCollectionServiceConfig,
+} from "../types";
+import { collectionSerializer } from "./serialziers";
+
+export const getCollectionDetailsInfra =
+  ({ apiURI }: SaleorCollectionServiceConfig): GetCollectionDetailsInfra =>
```
Comment (GrzegorzDerdak) [inline packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts:15]:
```suggestion
  ({ apiURI, logger }: SaleorCollectionServiceConfig): GetCollectionDetailsInfra =>
```

File: packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import { type LanguageCodeEnum } from "@nimara/codegen/schema";
+import { err, ok } from "@nimara/domain/objects/Result";
+
+import { graphqlClientV2 } from "#root/graphql/client";
+import { logger } from "#root/logging/service";
```
Comment (GrzegorzDerdak) [inline packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts:5]: Remove, and pass it in a `SaleorCollectionServiceConfig`

**Review by GrzegorzDerdak — CHANGES_REQUESTED** [review]

File: packages/infrastructure/src/public/saleor/collection/graphql/queries/CollectionDetailsQuery.graphql [hunk 5]
```diff
@@ -0,0 +1,13 @@
+query CollectionDetailsQuery(
+  $slug: String
```
[the remark this answers is not in the document]
Comment (gajda-w) [reply packages/infrastructure/src/public/saleor/collection/graphql/queries/CollectionDetailsQuery.graphql:2]: It's optional in the query, but I made it required when calling collectionService.getCollectionDetails so that ts prevents omitting the slug

File: apps/storefront/src/app/[locale]/(main)/collections/[slug]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+import { SearchPagination } from "../../_components/search-pagination";
+
+type SearchParams = Promise<{
+  after?: string;
+  before?: string;
+  limit?: string;
+}>;
+
+type Params = Promise<{
+  slug: string;
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/app/[locale]/(main)/collections/[slug]/page.tsx:22]: There's also a locale param in the path

File: packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            ...pageInfo,
+          },
+          operationName: "CollectionDetailsQuery",
+          options,
+        },
+      );
+
+      if (!result.ok) {
+        logger.error("Failed to fetch collection from Saleor", {
+          result,
```
Comment (GrzegorzDerdak) [inline packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts:50]: I discovered today that a better way is to do this to make the error message shorter so it's not cut out in Vercel logs.
```suggestion
          error: result.error,
```

**Review by GrzegorzDerdak — APPROVED** [review]
